src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java

```diff
@@ -166,10 +166,10 @@ private synchronized void setupMinecraft(ConfigContext configContext) throws Exc
 		extension.setMinecraftProvider(minecraftProvider);
 		minecraftProvider.provide();
 
-		// Created any layered mapping files.
+		final DependencyInfo mappingsDep = DependencyInfo.create(getProject(), Configurations.MAPPINGS);
+		// Created any layered mapping files. (After resolving the mappings dependency)
 		LayeredMappingsFactory.afterEvaluate(configContext);
```

Comment:
This doesnt seem right to me, LayeredMappingsFactory.afterEvaluate is what generates the layered mapping, but calling DependencyInfo before this has the chance of resolving the mappings before they have been created. I will have a look at this, the real fix is likely a bit more complex.